### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by worseTyler**

### CODE REVIEW

**Comment by cmartinezaraujo** [comment]
I will review this.

**Review by cmartinezaraujo — COMMENTED** [review]
Assignment

Jester class takes in the correct interfaces: ✔
Jester dependencies are null checked: ✔
Jester approprietly retrieves jokes and sends them to the output: ✔
Chuck Norris jokes ge filtered out: ✔
Jester code coverage is above 90%: ? I ran coverlet on your code but  to be honest I'm still not sure if i'm using the tool right. Here is a screen shot of the results it gave.

TC


Extra Credit

Output to console implementat is unit tested: ❌
JokeService uses JSON: ❌

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 1]
```diff
@@ -1,10 +1,68 @@
...
+
     [TestClass]
     public class JesterTests
     {
-        
+        [TestMethod]
+        public void TellJoke_WhenCalled_RunsBothInterfacedMethods()
+        {
+            Mock<IJokeService> iJokeService = new ();
+            Mock<IOutputJoke> iOutputJoke = new ();
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:14]: In general we don't include the "i" when naming variables
```suggestion
            Mock<IJokeService> jokeService = new ();
            Mock<IOutputJoke> outputJoke = new ();
```

File: CanHazFunny/CanHazFunny/IJokeService.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace CanHazFunny
+{
+    public interface IJokeService
+    {
+        public string GetJoke();
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny/IJokeService.cs:5]: For interfaces, avoid specifying access modifiers.
```suggestion
        string GetJoke();
```

File: CanHazFunny/CanHazFunny/IOutputJoke.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace CanHazFunny
+{
+    public interface IOutputJoke
+    {
+        public void OutputJoke(string joke);
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny/IOutputJoke.cs:5]:
```suggestion
        void OutputJoke(string joke);
```

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        public Jester(IJokeService jokeService, IOutputJoke outputJoke)
+        {
+            this.IJokeService = jokeService ?? throw new ArgumentNullException(nameof(this.IJokeService));
+            this.IOutputJoke = outputJoke ?? throw new ArgumentNullException(nameof(this.IOutputJoke));
+        }
+
+        private IJokeService IJokeService { get; }
+
+        private IOutputJoke IOutputJoke { get; }
```
Comment (Keboo) [inline CanHazFunny/CanHazFunny/Jester.cs:15]:
```suggestion
        private IJokeService JokeService { get; }

        private IOutputJoke OutputJoke { get; }
```

**Pull request closed without merging**